source/connect/connection-options.txt

```diff
@@ -0,0 +1,299 @@
+   client = Mongo::Client.new(uri)
+
+Using a ``Mongo::Client``
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can pass connection options as arguments to the ``Mongo::Client`` constructor
+instead of including them in your connection URI.
+When you configure your connection this way, it easier for you to
+change settings at runtime and catch errors during compilation.
+The following example shows how to use the ``Mongo::Client`` constructor to set
+connection options:
+
+.. code-block:: ruby
+
+   uri = "mongodb://<hostname>:<port>"
+   client = Mongo::Client.new(uri, connect_timeout: 60000, ssl: true)
+
+Connection Options
+------------------
+
+The following sections describe the connection options available in the {+driver-short+}.
+
+Network Compression
+~~~~~~~~~~~~~~~~~~~
+
+.. list-table::
+   :header-rows: 1
+   :widths: 30 70
+
+   * - Connection Option
+     - Description
+
+   * - **:compressors**
+     - | A list of potential compressors to use, in order of preference.
+         The driver chooses the first compressor that is also supported 
+         by the server. Currently the driver only supports ``zstd``, ``snappy``, and ``zlib``.
+       |
+       | **Data Type**: ``Array<String>``
+       | **Default**: none
+       | **Client Example**: ``compressors: ['snappy', 'zstd', 'zlib']``
+       | **Connection URI Example**: ``compressors=snappy,zstd,zlib``
+
+   * - **:zlib_compression_level**
+     - | The Zlib compression level to use, if using compression. 
+         This option accepts an integer value between ``-1`` and ``9``:
+       | 
+       | - **-1:** zlib uses its default compression level (usually ``6``).
+       | - **0:** No compression.
+       | - **1:** Fastest speed but lowest compression.
+       | - **9:** Best compression but slowest speed.
+       |
+       | For more information, see Ruby's `ZLib module <https://ruby-doc.org/stdlib-2.7.0/libdoc/zlib/rdoc/Zlib.html>`__ 
+         documentation.
+       | 
+       | **Data Type**: ``Integer``
+       | **Default**: ``None``
+       | **Client Example**: ``zlib_compression_level: 3``
+       | **Connection URI Example**: ``zlibCompressionLevel=3``
+
+Timeouts
+~~~~~~~~
+
+.. list-table::
+   :header-rows: 1
+   :widths: 30 70
+
+   * - Connection Option
+     - Description
+
+   * - **:connect_timeout**
+     - | The number of seconds to wait to establish a socket 
+         connection before raising an exception. This 
+         timeout is also used for SRV DNS record resolution. 
+       | 
+       | ``nil`` and ``0`` mean no timeout. Client creation will 
+         fail with an error if an invalid timeout value
+         is passed (such as a negative value or a non-numeric value).
+       |
+       | **Data Type**: ``Float``
+       | **Default**: ``10.0``
+       | **Client Example**: ``connect_timeout: 10.0``
+       | **Connection URI Example**: ``connectTimeoutMS=10000``
+
+   * - **:socket_timeout**
```

Comment:
I'm not sure that we want to document `socket_timeout`, as it's been deprecated. As of the implementation of RUBY-2523 (Client-Side Operations Timeout, or "CSOT"), many timeout options (like `socket_timeout` and `wait_queue_timeout`) are now deprecated and replaced with the single `timeout_ms` option.

The deprecated options are still supported; we just recommend moving to the `timeout_ms` option.